### ISSUE TITLE
GM_listValues does not work as of Firefox 35

### ISSUE THREAD

**Issue by Joe136** [greasemonkey/greasemonkey#2033]
"Support for exposing privileged objects to untrusted content via __exposedProps__ is being gradually removed"

The function "GM_listValues" doesn't work correctly for a few days in Firefox Nightly 35.0.

When I call it, I get an Array with the array length but without content. None of the functions ([i], forEach, some, every, toString) works, but the Web Console can show the content.

Now I have looked for the problem and found some points (including a workaround):
 1. if I type in <code>console.log(GM_listValues)</code>, I get this warning:
"Security wrapper denied access to property undefined on privileged Javascript object. Support for exposing privileged objects to untrusted content via \_\_exposedProps\_\_ is being gradually removed - use WebIDL bindings or Components.utils.cloneInto instead."
 1. in the Web Console: <code>GM_listValues()</code> shows me an empty array with the (correct, nonzero) length
 1. <code>console.log(GM_listValues())</code> shows me the array content, length and the \_\_exposedProps\_\_ object
 1. <code>var list = cloneInto(GM_listValues(), window);</code> works

It seems to be an increase of security in Firefox (and not a bug) so I don't think, <code>GM_listValues()</code> will work with the current implementation anymore.

Best Regards
Joe

**Comment by kuehlschrank** [comment]
Any progress?

Calling `abort()` on the object returned by `GM_xmlhttpRequest()` throws "Exposing privileged or cross-origin callable is prohibited" under Firefox 36 (now in beta channel).

**Comment by deltamarkus** [comment]
Hello GM_listValue don't work anymore
but someone has an fix for greasymonkey
see this link https://greasyfork.org/de/forum/discussion/2918/issue-with-gm-getvalue-from-gm-listvalues-array-in-firefox#latest
please update the greasemonkey version

thank you

**Comment by lukeIam** [comment]
@deltamarkus: Thanks a lot!
````JS
var list = cloneInto(GM_listValues(), window);
``` 
worked for my case.

**Comment by arantius** [comment]
I think this was addressed in Greasemonkey 3.0?  Please reopen if you still see issues.

**Comment by Joe136** [comment]
Great, it works on my Nightly version without e10.
